### PR TITLE
Super admin crud for libraries

### PR DESCRIPTION
**Pull request by KFad11**

### CODE REVIEW

File: app/models/member.rb [hunk 1]
```diff
@@ -6,6 +6,7 @@ class Member < ApplicationRecord
   has_many :loan_summaries
 
   has_many :holds, dependent: :destroy
+  has_many :libraries
```
Comment (benlangfeld) [inline app/models/member.rb]:
```suggestion
```

https://github.com/KFad11/circulate/pull/1/files#diff-99215219f23b3e704ba1b4ecadb52a71R45

File: app/controllers/libraries_controller.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  def destroy
+    @library = Library.find(params[:id])
+    @library.destroy
+    redirect_to libraries_path
+  end
+
+  private
+
+  def require_super_admin
+    # debugger
```
Comment (benlangfeld) [inline app/controllers/libraries_controller.rb]:
```suggestion
```

File: app/models/user.rb [hunk 3]
```diff
@@ -11,7 +11,8 @@ class User < ApplicationRecord
   enum role: {
     member: "member",
     staff: "staff",
-    admin: "admin"
+    admin: "admin",
+    super_admin: "super_admin"
```
Comment (benlangfeld) [inline app/models/user.rb]:
```suggestion
    super_admin: "super_admin",
```

This means that future diffs are simpler when adding new entries to this hash.

File: app/views/libraries/index.erb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<h1>Libraries Index</h1>
+
+<%= button_to "Add a new Library", new_library_path, method: :get %>
+<ul>
+  <% @libraries.each do |library| %>
+    <li><%= link_to library.name, library_path(library) %></li>
+  <% end %>
+</ul>
```
Comment (benlangfeld) [inline app/views/libraries/index.erb]: Please set your text editor to ensure terminating newlines.

File: app/controllers/libraries_controller.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
+class LibrariesController < ApplicationController
```
Comment (neildecapia) [inline app/controllers/libraries_controller.rb]: Should we have namespaced this inside `Admin`?

Comment (KFad11) [reply app/controllers/libraries_controller.rb]: I thought about this looking at their file structure. I was thinking the `libraries` and other `super_admin` files might even get their own namespace? Might be a Jim question.

Comment (benlangfeld) [reply app/controllers/libraries_controller.rb]: I vote for a SuperAdmin namespace, and then to merge this.

**Comment by KFad11** [comment]
@benlangfeld @neildecapia - This should be good to go code review wise, then to merge if everything looks OK. Let me know if we want to mess with the namespacing of the LibrariesController here, or doing something about it in a later PR. 

Also, let me know if it's necessary to clean up the commits at the end there. I committed the code review changes through github (`e4f10b8`), then committed the additional new lines from my text editor and terminal(`997318b`), but then had to fetch the code review changes to push the new lines commit due to being out of sync after doing that initial commit through github. So that gave me a duplicate commit (`880a4cb`) of the code review changes (`e4f10b8`).

I'm assuming if it needs all that explaination it's probably worth cleaning up... in that case I just hard reset back to commit `e4f10b8`, then `git fetch && git pull` to make sure my local version is up to date with that... THEN add the new lines and commit those changes and not get the duplicate commit after?

**Comment by neildecapia** [comment]
@KFad11 let's settle the namespace here. I'm sure there's more work left on this so it would be nice to get this in a good place from the start.

**Comment by benlangfeld** [comment]
Let's get the SuperAdmin namespace in place, then squash-merge this into https://github.com/rubyforgood/circulate/pull/215

**Comment by KFad11** [comment]
@benlangfeld - Anywhere else the namespace needs to be applied? The tests perhaps?

File: config/routes.rb [hunk 6]
```diff
@@ -104,6 +104,7 @@
     resource :password, only: [:edit, :update]
   end
 
+  resources :libraries
```
Comment (benlangfeld) [inline config/routes.rb]: This needs namespacing.

**Comment by benlangfeld** [comment]
Looks like you didn't include tests for this. You should; follow the model of some of the normal admin controller tests.

**Comment by neildecapia** [comment]
@KFad11 @benlangfeld I've updated the tests [here](https://github.com/KFad11/circulate/pull/3).

**Pull request merged**